common/spec/dependabot/shared_helpers_spec.rb

```diff
@@ -314,4 +314,140 @@
       )
     end
   end
+
+  describe ".with_git_configured" do
+    config_header = "Generated by dependabot/dependabot-core"
+
+    credentials_helper = <<~CONFIG.chomp
+      [credential]
+      	helper = !#{Dependabot::SharedHelpers.credential_helper_path} --file #{Dir.pwd}/git.store
+    CONFIG
+
+    def alternatives(host)
+      <<~CONFIG.chomp
+        [url "https://#{host}/"]
+        	insteadOf = ssh://git@#{host}/
+        	insteadOf = ssh://git@#{host}:
+        	insteadOf = git@#{host}:
+        	insteadOf = git@#{host}/
+        	insteadOf = git://#{host}/
+      CONFIG
+    end
+
+    let(:credentials) { [] }
+
+    subject(:with_git_configured) do
+      Dependabot::SharedHelpers.with_git_configured(credentials: credentials) do
+        [git_config.call, git_credentials.call]
+      end
+    end
+
+    let(:git_config) { -> { `cat ~/.gitconfig` } }
+    let(:git_credentials) { -> { `cat #{Dir.pwd}/git.store` } }
+
+    context "when providing no extra credentials" do
+      let(:credentials) { [] }
+
+      it "creates a .gitconfig that contains the Dependabot header" do
+        expect(with_git_configured[0]).to include(config_header)
```

Comment:
It's not immediately clear to be what the first element of `with_git_configured` is, could we pull out a `let` that describes it?

```ruby
let(:configured_git_config) { with_git_configured.first }
let(:configured_git_credentials) { with_git_configured.last }
```

Or even two separate `let`s, they don't _need_ to be a `subject`

```ruby
let(:configured_git_config) do
  Dependabot::SharedHelpers.with_git_configured(credentials: credentials) do
    `cat ~/.gitconfig`
  end
end
```

Reply:
Yeah makes sense. What do you think of this:

```go
def with_git_configured(&block)
  Dependabot::SharedHelpers.with_git_configured(credentials: credentials) { block.call }
end

let(:configured_git_config) { with_git_configured { `cat ~/.gitconfig` } }
let(:configured_git_credentials) { with_git_configured { `cat #{Dir.pwd}/git.store` } }
```

Reply:
I dig it 👍